src/perlin.h

```diff
@@ -0,0 +1,163 @@
+#ifndef RAYTRACING_PERLIN_H
+#define RAYTRACING_PERLIN_H
+
+
+#include "vec3.h"
+#include "ray.h"
+
+
+class Perlin
+{
+
+private:
+    static int *perm_x;
+    static int *perm_y;
+    static int *perm_z;
+    static float *ranfloat;
+
+public:
+
+    /**
+     * Noise processing function.
+     *
+     * This function gives the noise at the given point in space.
+     *
+     * @param p The space coordinate to use for calculating the Perlin noise value.
+     *
+     * @return The scalar value for the perlin noise.
+     */
+    float noise(const Vec3 &p) const;
+
+};
+
+
+/**
+ * Trilinear interpolation function.
+ *
+ * @todo This function may be better placed into the texture.h header file.
+ *
+ * @param c
+ * @param u
+ * @param v
+ * @param w
+ *
+ * @return
+ */
+inline float trilinear_interpolation(float c[2][2][2], float u, float v, float w)
```

Comment:
Are we sure that this function is good to be `inline`?

Reply:
Don't know, why should I change to "not" be an `inline` function?

Reply:
> Because of the potential for code bloat, inlining a function is best suited to short functions (e.g. no more than a few lines) that are typically called inside loops and do not branch. Also note that the inline keyword is only a recommendation -- the compiler is free to ignore your request to inline a function. This is likely to be the result if you try to inline a lengthy function!
> 
> Finally, modern compilers are now very good at inlining functions automatically -- better than humans in most cases. Even if you don’t mark a function as inline, the compiler will inline functions that it believes will result in performance increases. Thus, in most cases, there isn’t a specific need to use the inline keyword. Let the compiler handle inlining functions for you.

Reply:
Just because of that!

Reply:
Will remove `inline` then. Cool.